### ISSUE TITLE
ACPI path and value errors in OpCore Simplify

### ISSUE THREAD

**Issue by Soupolait** [lzhoang2801/Hardware-Sniffer#49]
Hey, while report and ACPI dump and hardware report go smoothly, importing said report in OpCore Simplify gives these errors : 
```
Errors (2):
    1. Root.System Devices.PNP0C14_#1.ACPI Path: Value '\AOD_' does not match pattern '^[\\]?_SB(\.[A-Z0-9_]+)+$'
    2. Root.System Devices.PNP0103.ACPI Path: Value '\HPET' does not match pattern '^[\\]?_SB(\.[A-Z0-9_]+)+$'

Warnings (7):
    1. Root.BIOS: Unknown key 'Above 4G Decoding'
    2. Root.GPU.NVIDIA Corporation GA107M [GeForce RTX 3050 Mobile]: Unknown key 'Bus Type'
    3. Root.GPU.Advanced Micro Devices, Inc. [AMD/ATI] Cezanne [Radeon Vega Series / Radeon Vega Mobile Series]: Unknown key 'Bus Type'
    4. Root.Sound.Realtek ALC294: Unknown key 'PCI Path'
    5. Root.Sound.Realtek ALC294: Unknown key 'ACPI Path'
    6. Root.Sound.Advanced Micro Devices, Inc. [AMD/ATI] Renoir/Cezanne HDMI/DP Audio Controller: Unknown key 'PCI Path'
    7. Root.Sound.Advanced Micro Devices, Inc. [AMD/ATI] Renoir/Cezanne HDMI/DP Audio Controller: Unknown key 'ACPI Path'
```
Ubuntu 25.10, AMD iGPU, NVIDIA dGPU.

Here are some relevant outputs : 
```
loic@Vivobook:~$ cat /sys/bus/pci/devices/*/firmware_node/path
\_SB_.PCI0.D003
\_SB_.PCI0.IOMA
\_SB_.PCI0.P010
\_SB_.PCI0.GPP0
\_SB_.PCI0.P020
\_SB_.PCI0.GPP4
\_SB_.PCI0.GPP6
\_SB_.PCI0.P080
\_SB_.PCI0.GP17
\_SB_.PCI0.GP18
\_SB_.PCI0.D01C
\_SB_.PCI0.SBRG
\_SB_.PCI0.P180
\_SB_.PCI0.P181
\_SB_.PCI0.P182
\_SB_.PCI0.P183
\_SB_.PCI0.P184
\_SB_.PCI0.P185
\_SB_.PCI0.P186
\_SB_.PCI0.P187
\_SB_.PCI0.GPP0.PEGP
\_SB_.PCI0.GPP4.WLAN
\_SB_.PCI0.GPP6.NVME
\_SB_.PCI0.GP17.VGA_
\_SB_.PCI0.GP17.HDAU
\_SB_.PCI0.GP17.APSP
\_SB_.PCI0.GP17.XHC0
\_SB_.PCI0.GP17.XHC1
\_SB_.PCI0.GP17.ACP_
\_SB_.PCI0.GP17.AZAL
\_SB_.PCI0.GP18.SATA
\_SB_.PCI0.GP18.SAT1
```
```
loic@Vivobook:~$ cat /sys/bus/acpi/devices/*/path
\_SB_.PCI0.SBRG.EC0_.AC0_
\_SB_.PLTF.P000
\_SB_.PLTF.P001
\_SB_.PLTF.P002
\_SB_.PLTF.P003
\_SB_.PLTF.P004
\_SB_.PLTF.P005
\_SB_.PLTF.P006
\_SB_.PLTF.P007
\_SB_.PLTF.P008
\_SB_.PLTF.P009
\_SB_.PLTF.P00A
\_SB_.PLTF.P00B
\_SB_.PLTF.P00C
\_SB_.PLTF.P00D
\_SB_.PLTF.P00E
\_SB_.PLTF.P00F
\_SB_.APAD
\_SB_.PLTF
\_SB_.PEP_
\_SB_.I2CA
\_SB_.I2CB
\_SB_.I2CC
\_SB_.I2CD
\_SB_.I2CE
\_SB_.I2CF
\_SB_.FUR0
\_SB_.FUR1
\_SB_.FUR2
\_SB_.FUR3
\_SB_.GPIO
\_SB_.EMM0
\_SB_.HFP1
\_SB_.I2CD.ETPD
\_SB_.ASUP
\_SB_.PCI0.SBRG.PS2K
\_SB_.PCI0.D003
\_SB_.PCI0.GPP1
\_SB_.PCI0.GPP1.DEV0
\_SB_.PCI0.GPP1.DEV1
\_SB_.PCI0.GPP1.D006
\_SB_.PCI0.GPP2
\_SB_.PCI0.GPP2.D008
\_SB_.PCI0.GPP3
\_SB_.PCI0.GPP3.D00A
\_SB_.PCI0.GPP4
\_SB_.PCI0.GPP4.WLAN
\_SB_.PCI0.GP17
\_SB_.PCI0.GP17.VGA_.LCD_
\_SB_.PCI0.GP17.HDAU
\_SB_.PCI0.GP17.ACP_
\_SB_.PCI0.GP17.AZAL
\_SB_.PCI0.GP17.XHC0
\_SB_.PCI0.GP17.XHC0.RHUB
\_SB_.PCI0.GP17.XHC0.RHUB.PRT1
\_SB_.PCI0.GP17.XHC0.RHUB.PRT2
\_SB_.PCI0.GP17.XHC0.RHUB.PRT3
\_SB_.PCI0.GP17.XHC0.RHUB.PRT4
\_SB_.PCI0.GP17.XHC0.RHUB.PRT5
\_SB_.PCI0.GP17.XHC0.RHUB.PRT6
\_SB_.PCI0.GP17.XHC1
\_SB_.PCI0.GP17.XHC1.RHUB
\_SB_.PCI0.GP17.XHC1.RHUB.PRT1
\_SB_.PCI0.GP17.XHC1.RHUB.PRT2
\_SB_.PCI0.GP17.XHC1.RHUB.PRT3
\_SB_.PCI0.GP17.XHC1.RHUB.PRT3.DCAM
\_SB_.PCI0.GP17.XHC1.RHUB.PRT4
\_SB_.PCI0.GP17.XHC1.RHUB.PRT5
\_SB_.PCI0.GP17.XHC1.RHUB.PRT6
\_SB_.PCI0.GP18
\_SB_.PCI0.GP18.SATA
\_SB_.PCI0.GP18.SAT1
\_SB_.PCI0.GP19
\_SB_.PCI0.GP19.D022
\_SB_.PCI0.GP19.D023
\_SB_.PCI0.GP19.D024
\_SB_.PCI0.D01C
\_SB_.PCI0.SBRG
\_SB_.PCI0.D01F
\_SB_.PCI0.GPP6
\_SB_.PCI0.GPP6.NVME
\_SB_.PCI0.GPP0
\_SB_.PCI0.GPP0.D004
\_SB_.PCI0.GPP0.PEGP
\_SB_.PCI0.GPP0.HDAU
\_SB_.PCI0.GPP5
\_SB_.PCI0.GPP5.D00C
\_SB_.PCI0.P010
\_SB_.PCI0.P020
\_SB_.PCI0.P080
\_SB_.PCI0.P180
\_SB_.PCI0.P181
\_SB_.PCI0.P182
\_SB_.PCI0.P183
\_SB_.PCI0.P184
\_SB_.PCI0.P185
\_SB_.PCI0.P186
\_SB_.PCI0.P187
\_SB_.PCI0.GP17.XHC0.P0U0
\_SB_.PCI0.GP17.XHC0.P3U0
\_SB_.PCI0.GP17.XHC1.P0U1
\_SB_.PCI0.GP17.XHC1.P3U1
\_SB_.PCI0.GP17.XHC1.RHUB.PRT2.BTPR
\_SB_.PCI0.GP18.SATA.P0SA
\_SB_.PCI0.GP18.SAT1.P0SA
\_SB_.PCI0.GPP0.M237
\_SB_.PRWL
\_SB_
\_TZ_
\
\_TZ_.THRM
\_SB_.PCI0.GP17.VGA_
\_SB_.TPM_
\_SB_.NPCF
\_SB_.PCI0.SBRG.PIC_
\_SB_.PCI0.SBRG.TMR_
\HPET
\_SB_.PCI0.SBRG.DMAD
\_SB_.PCI0.UAR1
\_SB_.PCI0.UAR2
\_SB_.PCI0.UAR3
\_SB_.PCI0.UAR4
\_SB_.PCI0.SBRG.SPKR
\_SB_.PCI0
\_SB_.PCI0.SBRG.RTC0
\_SB_.PCI0.AMDN
\_SB_.PCI0.IOMA
\_SB_.PCI0.GP17.APSP
\_SB_.PCI0.S900
\_SB_.AWR0
\_SB_.AWR0.ABR0
\_SB_.AWR0.ABR1
\_SB_.AWR0.ABR2
\_SB_.AWR0.ABR3
\_SB_.AWR0.ABR4
\_SB_.AWR0.ABR5
\_SB_.AWR0.ABR6
\_SB_.AWR0.ABR7
\_SB_.AWR0.ABR8
\_SB_.AWR0.ABR9
\_SB_.PCI0.SBRG.EC0_
\_SB_.PCI0.SBRG.EC0_.BAT0
\_SB_.PWRB
\_SB_.PCI0.SBRG.EC0_.LID_
\_SB_.SLPB
\_SB_.LNKA
\_SB_.LNKB
\_SB_.LNKC
\_SB_.LNKD
\_SB_.LNKE
\_SB_.LNKF
\_SB_.LNKG
\_SB_.LNKH
\_SB_.ATKD
\AOD_
```

Also, here is the [report](https://github.com/user-attachments/files/24278903/Report.zip) from Hardware Sniffer.

**Comment by Flexyz0** [comment]
either im blind or nothing makes sense

can you explain the steps to reproduce, what did you do step by step until you gave report to OpCore-Simplify

**Comment by Flexyz0** [comment]
also did you installed ubuntu 25.10 yourself or it was installed on this device when it came to you

**Comment by Soupolait** [comment]
Sure.
```
git clone https://github.com/lzhoang2801/Hardware-Sniffer.git
cd Hardware-Sniffer
python3 HardwareSniffer.py
H #Hardware Report
A #ACPI dump
q #Quit
cd ../

git clone https://github.com/lzhoang2801/OpCore-Simplify.git
cd OpCore-Simplify
python3 OpCore-Simplify.py
1 #Select Hardware Report
/home/loic/Hardware-Sniffer/SysReport/Report.json
```
Then I got the errors listed above.
I installed Ubuntu myself.

**Comment by Flexyz0** [comment]
okay thats interesting but i think i figured it out can you run this and share output

**Comment by Soupolait** [comment]
Hey, I'm met with those errors :when running the last file you sent : 
```
loic@Vivobook:~/Téléchargements/OpCore-Simplify$ python3 test.py 
/home/loic/Téléchargements/OpCore-Simplify/test.py:8: SyntaxWarning: invalid escape sequence '\_'
  if not f.startswith('\_SB'):
Error opening PNP0C14:01: [Errno 21] Is a directory: '/sys/bus/acpi/devices/PNP0C14:01/wakeup'
Error opening LNXTHERM:00: [Errno 21] Is a directory: '/sys/bus/acpi/devices/LNXTHERM:00/thermal_zone'
Error opening LNXSYBUS:01: [Errno 21] Is a directory: '/sys/bus/acpi/devices/LNXSYBUS:01/power'
Error opening PNP0103:00: [Errno 21] Is a directory: '/sys/bus/acpi/devices/PNP0103:00/power'
Error opening LNXSYSTM:00: [Errno 21] Is a directory: '/sys/bus/acpi/devices/LNXSYSTM:00/power'
('PNP0C14:01', '\\AOD_')
('LNXTHERM:00', '\\_TZ_.THRM')
('LNXSYBUS:01', '\\_TZ_')
('PNP0103:00', '\\HPET')
('LNXSYSTM:00', '\\')
```
I wasn't sure so I tried in HS folder and OCS folder.

**Comment by Flexyz0** [comment]
last files i uploaded were not what i wanted this one should make much sense

[test.py](https://github.com/user-attachments/files/24280658/test.py)

**Comment by Soupolait** [comment]
My bad. Here's the output :
```
loic@Vivobook:~/Téléchargements/OpCore-Simplify$ python3 test.py 
/home/loic/Téléchargements/OpCore-Simplify/test.py:8: SyntaxWarning: invalid escape sequence '\_'
  if not f.startswith('\_SB'):
('PNP0C14:01', {'uevent': 'MODALIAS=acpi:PNP0C14:', 'uid': 'AOD', 'hid': 'PNP0C14', 'path': '\\AOD_', 'modalias': 'acpi:PNP0C14:'})
('LNXTHERM:00', {'uevent': 'DRIVER=thermal\nMODALIAS=acpi:LNXTHERM:', 'hid': 'LNXTHERM', 'path': '\\_TZ_.THRM', 'modalias': 'acpi:LNXTHERM:'})
('LNXSYBUS:01', {'uevent': 'MODALIAS=acpi:LNXSYBUS:', 'hid': 'LNXSYBUS', 'path': '\\_TZ_', 'modalias': 'acpi:LNXSYBUS:'})
('PNP0103:00', {'uevent': 'MODALIAS=acpi:PNP0103:', 'hid': 'PNP0103', 'path': '\\HPET', 'status': '15', 'modalias': 'acpi:PNP0103:'})
('LNXSYSTM:00', {'uevent': 'MODALIAS=acpi:LNXSYSTM:', 'hid': 'LNXSYSTM', 'path': '\\', 'modalias': 'acpi:LNXSYSTM:'})
```

**Comment by Flexyz0** [comment]
okay apparently HPET is a acpi table which is specifically used in arm64 x86 processors shortly newer models of amd processors which will not cause any problem to you but i couldnt find anything related to AOD acpi which is the one that doesn't makes sense this device found 2 times by script with 2 different acpi table

**Comment by Flexyz0** [comment]
if you want to try installing hackintosh these 2 shouldnt cause you any problem if you just delete them manually from your Report.json but im not the one advising this to you installing hackintosh always your own responsibility.

if you want you can delete `"PNP0C14_#1": { ... },` from line 230 to 234 in your report.json and also you can remove `"ACPI Path": "\\HPET"` in line 253. Only deleting ACPI line would be the safest

**Comment by Soupolait** [comment]
Thanks !
